phoenix-core/src/main/java/org/apache/phoenix/expression/function/RoundJodaDateExpression.java

```diff
@@ -48,9 +49,10 @@ public boolean evaluate(Tuple tuple, ImmutableBytesWritable ptr) {
             }
             PDataType dataType = getDataType();
             long time = dataType.getCodec().decodeLong(ptr, children.get(0).getSortOrder());
-            DateTime dt = new DateTime(time,ISOChronology.getInstanceUTC());
+            int offset = Calendar.getInstance().getTimeZone().getRawOffset();
+            DateTime dt = new DateTime(time + offset,ISOChronology.getInstanceUTC());
             long value = roundDateTime(dt);
-            Date d = new Date(value);
+            Date d = new Date(value - offset);
```

Comment:
In RoundDate expression offset was handled inside of the rounding function.  I'd either recommend the same pattern or maybe handling this in a utility class?